plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeManager.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.exchange;
+
+import io.trino.spi.TrinoException;
+import io.trino.spi.exchange.Exchange;
+import io.trino.spi.exchange.ExchangeContext;
+import io.trino.spi.exchange.ExchangeManager;
+import io.trino.spi.exchange.ExchangeSink;
+import io.trino.spi.exchange.ExchangeSinkInstanceHandle;
+import io.trino.spi.exchange.ExchangeSource;
+import io.trino.spi.exchange.ExchangeSourceHandle;
+
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.inject.Inject;
+
+import java.net.URI;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static java.util.Objects.requireNonNull;
+
+public class FileSystemExchangeManager
+        implements ExchangeManager
+{
+    public static final String AES = "AES";
+    public static final int KEY_BITS = 256;
+    public static final String PATH_SEPARATOR = "/";
+
+    private final FileSystemExchangeStorage exchangeStorage;
+    private final URI baseDirectory;
+    private final boolean exchangeEncryptionEnabled;
+
+    @Inject
+    public FileSystemExchangeManager(FileSystemExchangeStorage exchangeStorage, FileSystemExchangeConfig fileSystemExchangeConfig)
+    {
+        this.exchangeStorage = requireNonNull(exchangeStorage, "exchangeStorage is null");
+        String baseDirectory = requireNonNull(fileSystemExchangeConfig.getBaseDirectory(), "baseDirectory is null");
```

Comment:
Can we make `fileSystemExchangeConfig.getBaseDirectory()` already return URI?

Reply:
Actually I tried and I found it really weird to do so, as I don't want to give a default value to base directory, and don't  want to involve too much logic in config getter function

Reply:
You can do sth like:
```java
    private Optional<URI> baseDirectory = Optional.empty();

    public Optional<URI> getBaseDirectory()
    {
        return baseDirectory;
    }

    @Config("exchange.base-directory")
    public FileSystemExchangeConfig setBaseDirectory(String baseDirectory)
    {
        this.baseDirectory = Optional.ofNullable(baseDirectory).map(URI::create);
        return this;
    }

```

We do alike quite often. But I do not feel strongly.